cartography/intel/aws/ec2/instances.py

```diff
@@ -21,410 +29,343 @@
+        reservation_list.append({
+            'RequesterId': reservation.get('RequesterId'),
+            'ReservationId': reservation['ReservationId'],
+            'OwnerId': reservation['OwnerId'],
+        })
+        for instance in reservation['Instances']:
+            instance_id = instance['InstanceId']
+            launch_time = instance.get("LaunchTime")
+            launch_time_unix = str(time.mktime(launch_time.timetuple())) if launch_time else None
+            instance_list.append(
+                {
+                    'InstanceId': instance_id,
+                    'ReservationId': reservation_id,
+                    'PublicDnsName': instance.get("PublicDnsName"),
+                    'PublicIpAddress': instance.get("PublicIpAddress"),
+                    'PrivateIpAddress': instance.get("PrivateIpAddress"),
+                    'ImageId': instance.get("ImageId"),
+                    'InstanceType': instance.get("InstanceType"),
+                    'IamInstanceProfile': instance.get("IamInstanceProfile", {}).get("Arn"),
+                    'MonitoringState': instance.get("Monitoring", {}).get("State"),
+                    'LaunchTime': instance.get("LaunchTime"),
+                    'LaunchTimeUnix': launch_time_unix,
+                    'State': instance.get("State", {}).get("Name"),
+                    'AvailabilityZone': instance.get("Placement", {}).get("AvailabilityZone"),
+                    'Tenancy': instance.get("Placement", {}).get("Tenancy"),
+                    'HostResourceGroupArn': instance.get("Placement", {}).get("HostResourceGroupArn"),
+                    'Platform': instance.get("Platform"),
+                    'Architecture': instance.get("Architecture"),
+                    'EbsOptimized': instance.get("EbsOptimized"),
+                    'BootMode': instance.get("BootMode"),
+                    'InstanceLifecycle': instance.get("InstanceLifecycle"),
+                    'HibernationOptions': instance.get("HibernationOptions", {}).get("Configured"),
+                },
+            )
 
+            subnet_id = instance.get('SubnetId')
+            if subnet_id:
+                subnet_list.append(
+                    {
+                        'SubnetId': subnet_id,
+                        'InstanceId': instance_id,
+                    },
+                )
 
-@timeit
-def load_ec2_instance_network_interfaces(neo4j_session: neo4j.Session, instance_data: Dict, update_tag: int) -> None:
-    neo4j_session.write_transaction(_load_ec2_instance_net_if_tx, instance_data, update_tag)
+            if instance.get("KeyName"):
+                key_name = instance["KeyName"]
+                key_pair_arn = f'arn:aws:ec2:{region}:{current_aws_account_id}:key-pair/{key_name}'
+                keypair_list.append({
+                    'KeyPairArn': key_pair_arn,
+                    'KeyName': key_name,
+                    'InstanceId': instance_id,
```

Comment:
Strange to set an instanceId here, since 1 keypair can be used with many instances, but it will only be used for the TargetNodeMatcher, and duplicates keypair dicts will still be merged to onto one node.

Reply:
Good callout, I'll look more closely.

Reply:
We talked about this a bit offline - I think this will be solved when we refactor the keypair sync to the new model

Additioanlly, an EC2 instance schema itself only defines a single EC2 keypair so I think we're good here: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_Instance.html